tests/5.1/metadirective/test_metadirective_target_device_num.c

```diff
@@ -0,0 +1,62 @@
+//===--------------------- test_metadirective_target_device_num.c ---------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+// 
+// Tests that target_device is recognized & will work properly when there is
+// an available target device, and that device number is 0.
+//
+////===---------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_metadirective_target_device() {
+   int errors = 0;
+   int A[N];
+
+   for (int i = 0; i < N; i++) {
+      A[i] = 1;
+   }
+
+   #pragma omp target map(tofrom: A)
+   {
+      #pragma omp parallel num_threads(4)
```

Comment:
remove num_threads -- > then we will get 128 threads for NVIDIA GPUs and 256 for AMD. Need to check if this holds for GNU.

Reply:
Also, what happens when this runs on host ?